app/models/manageiq/providers/redfish/inventory/parser/physical_infra_manager.rb

```diff
@@ -28,5 +29,36 @@ def physical_servers
         )
       end
     end
+
+    def physical_server_details
+      # TODO(tadeboro): There is no similar data in Redfish service, so
+      # mapping will need to be quite sophisticated if we would like to get
+      # more info into database.
+      collector.physical_server_details.each do |d|
+        server = persister.physical_servers.find_or_build(d[:server_id])
+        details = persister.physical_server_details.find_or_build(server)
+        details.assign_attributes(
```

Comment:
Is it possible for the server details to be built anywhere else?  If these can't be created from multiple code paths it'd be better to skip the find_or_build and just use build.

Reply:
Server details are only built here. I used `find_or_build` here because I though that calling `build` might introduce duplicated entries on inventory refresh. Of course, I may be completely wrong here.

Reply:
Ah okay I see what you mean, no the find_or_build part is only relative to the current data in the inventory collection, not what is in the database...that is evaluated during save time.

What you can also do is instead of `server = find_or_build(..)` you can do a `server = persister.physical_servers.lazy_find(d[:server_id])` which will allow the server link to be evaluated at save time even if the server isn't in your current inventory_collections aka you were doing a targeted refresh.

Reply:
I updated the code with your recommendations (and made a mental note to read the inventory-related source code once again).